includes/alerts.inc.php

```diff
@@ -879,3 +884,25 @@ function ExtTransports($obj)
         echo '; ';
     }
 }//end ExtTransports()
+
+/**
+ * Check if a device's all parent are down
+ * Returns true if all parents are down
+ * @param int $device Device-ID
+ * @return bool
+ */
+function IsParentDown($device)
+{
+    $parent_count = dbFetchCell("SELECT count(`parent_device_id`) from `device_relationships` WHERE `child_device_id` = ?", array($device));
+    if (!$parent_id || $parent_id == null || $parent_id == 0) {
```

Comment:
Might be this that's causing it. Not checking $parent_count here but $parent_id.

Reply:
Yeah, should be:
```
if (!$parent_count) {
    return false;
}
```
that covers 0 and null btw.